Sources/KituraNet/URLParser.swift

```diff
@@ -22,29 +22,89 @@ import Foundation
 
 public class URLParser: CustomStringConvertible {
 
+    private var urlComponents: URLComponents?
+
     /// Schema.
-    public var schema: String?
+    public var schema: String? {
+        get {
+            return self.urlComponents?.scheme
+        }
+        set (newValue) {
+            self.urlComponents?.scheme = newValue
+        }
+    }
 
     /// Hostname.
-    public var host: String?
+    public var host: String? {
+        get {
+            return self.urlComponents?.host
+        }
+        set (newValue) {
+            self.urlComponents?.host = newValue
+        }
+    }
 
     /// Path portion of the URL.
-    public var path: String?
+    public var path: String? {
+        get {
+            return self.urlComponents?.path
+        }
+        set (newValue) {
+            if let newValue = newValue {
+                self.urlComponents?.path = newValue
+            }
+        }
+    }
 
     /// The entire query portion of the URL.
-    public var query: String?
+    public var query: String? {
+        get {
+            return self.urlComponents?.query
+        }
+        set (newValue) {
+            self.urlComponents?.query = newValue
+        }
+    }
 
     /// An optional fragment identifier providing direction to a secondary resource.
-    public var fragment: String?
+    public var fragment: String? {
+        get {
+            return self.urlComponents?.fragment
+        }
+        set (newValue) {
+            self.urlComponents?.fragment = newValue
+        }
+    }
 
     /// The userid and password if specified in the URL.
-    public var userinfo: String?
+    public var userinfo: String? {
+        var _userInfo: String? = nil
+        if let username = urlComponents?.user, let password = urlComponents?.password {
+            _userInfo = "\(username):\(password)"
+        }
+        return _userInfo
+    }
 
     /// The port specified, if any, in the URL.
-    public var port: Int?
+    public var port: Int? {
+        get {
+            return self.urlComponents?.port
+        }
+        set (newValue) {
+            self.urlComponents?.port = newValue
+        }
+    }
 
     /// The query parameters broken out.
-    public var queryParameters: [String: String] = [:]
+    public var queryParameters: [String: String] {
+        var _query: [String: String] = [:]
```

Comment:
This will calculate the `queryParameters` dictionary every time the property is accessed. For this property we should cache the dictionary.

Reply:
@ianpartridge I agree! Thanks for pointing this out :)